### PR TITLE
Validate inputs for cropping

### PR DESCRIPTION
**Pull request by ngreenwald**
Adds more thorough checking for keywords for crop function

### CODE REVIEW

File: caliban_toolbox/reshape_data.py [hunk 1]
```diff
@@ -54,12 +54,33 @@ def crop_multichannel_data(X_data, y_data, crop_size=None, crop_num=None, overla
...
+    if crop_size is not None:
+        if not isinstance(crop_size, tuple):
+            raise ValueError('crop_size must be a tuple')
+
+        if len(crop_size) != 2:
+            raise ValueError('crop_size must be a tuple of (row_crop, col_crop), '
+                             'got {}'.format(crop_size))
+
+        if not crop_size[0] > 0 and crop_size[1] > 0:
+            raise ValueError('crop_size entries must be positive numbers')
```
Comment (geneva-miller) [inline caliban_toolbox/reshape_data.py:72]: should we restrict this specifically to positive integers?

File: caliban_toolbox/reshape_data.py [hunk 2]
```diff
@@ -54,12 +54,33 @@ def crop_multichannel_data(X_data, y_data, crop_size=None, crop_num=None, overla
...
+    if crop_num is not None:
+        if not isinstance(crop_num, tuple):
+            raise ValueError('crop_num must be a tuple')
+
+        if len(crop_num) != 2:
+            raise ValueError('crop_num must be a tuple of (num_row, num_col), '
+                             'got {}'.format(crop_size))
+
+        if not crop_num[0] > 0 and crop_num[1] > 0:
+            raise ValueError('crop_num entries must be positive numbers')
```
Comment (geneva-miller) [inline caliban_toolbox/reshape_data.py:83]: positive integers? ie, change wording here but also check `int`

Comment (ngreenwald) [reply caliban_toolbox/reshape_data.py]: Good idea

File: caliban_toolbox/reshape_data.py [hunk 3]
```diff
@@ -54,12 +54,39 @@ def crop_multichannel_data(X_data, y_data, crop_size=None, crop_num=None, overla
...
+        raise ValueError('Either crop_size or crop_num must be specified')
 
-    if not crop_size[0] > 0 and crop_size[1] > 0:
-        raise ValueError('crop_size entries must be positive numbers')
+    if crop_size is not None and crop_num is not None:
+        raise ValueError('Only one of crop_size and crop_num should be provided')
+
+    if crop_size is not None:
+        if not isinstance(crop_size, tuple):
+            raise ValueError('crop_size must be a tuple')
```
Comment (willgraf) [inline caliban_toolbox/reshape_data.py:65]: It can't be a list?

**Pull request merged**